terraform/ssh/ssh.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package ssh
+
+import (
+	"io"
+	"net"
+	"os"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+// Conn is a wrapper type over a ssh connection
+// that takes care of creating a channel and running
+// commands in a single method.
+type Conn struct {
+	client *ssh.Client
+}
+
+// NewConn returns a Conn object by dialing
+// to the local ssh agent.
+func NewConn(ip string) (*Conn, error) {
+	conn, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
+	if err != nil {
+		return nil, err
+	}
+
+	config := &ssh.ClientConfig{
+		User: "ubuntu",
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeysCallback(agent.NewClient(conn).Signers),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	sshc, err := ssh.Dial("tcp", ip+":22", config)
+	if err != nil {
+		return nil, err
+	}
+	return &Conn{client: sshc}, nil
+}
+
+// RunCommand runs a given command in a new ssh session.
+func (sshc *Conn) RunCommand(cmd string) error {
+	sess, err := sshc.client.NewSession()
+	if err != nil {
+		return err
+	}
+	defer sess.Close()
```

Comment:
I gave up on error checking the Close method. It was just becoming too cumbersome. It needed a named param, because I would need to modify the returned error, and also at the same time check if sess.Run has returned an error or not. And then it would get more complicated, because now I need to decide what should I return ? The error from Run or the one from Close ? Or maybe both ?

In the end, I just decided to drop it.